src/stores/GuildMemberRoleStore.js

```diff
@@ -138,6 +149,23 @@ class GuildMemberRoleStore extends Collection {
    *   .catch(console.error);
    */
   set(roles, reason) {
+    // eslint-disable-next-line eqeqeq
+    if (roles == undefined) return Promise.resolve(this.member);
+
+    if (!(roles instanceof Collection)) {
+      if (!Array.isArray(roles)) {
+        return Promise.reject(new TypeError('INVALID_TYPE', 'roles',
+          'Array or Collection of Roles or Snowflakes', true));
+      }
+
+      roles = roles.map(role => [role, this.guild.roles.resolve(role)]);
```

Comment:
I don't want to be *that guy*.. but I will.. This code does 4 (or probably more) iterations. Mapping the array/collection to a `[string | Role, Role | null][]`, running a some check on that, creating another collection ON that, then filtering AGAIN and concatenating with another filter. I would think this might be cleaner and simpler code, but please do test before using!

```js
set(roles, reason) {
    // eslint-disable-next-line eqeqeq
    if (roles == undefined) return Promise.resolve(this.member);
	
	if (roles instanceof Collection) roles = [...roles.values()];
	if (!Array.isArray(roles)) {
		return Promise.reject(new TypeError('INVALID_TYPE', 'roles',
          'Array or Collection of Roles or Snowflakes', true));
	}

	roles = roles.map(this.guild.roles.resolve);
	
	if (roles.includes(null)) {
		return Promise.reject(new TypeError('INVALID_TYPE', 'roles',
          'Array or Collection of Roles or Snowflakes', true));
	}

	roles = roles.filter(role => !role.managed).concat(this.filter(role => role.managed));

	return this.member.edit({ roles }, reason);
}
```

Reply:
I would have prefered to use straight arrays instead like this code you gave me shows, however I was told in the support server that collections were preferred so i went with that instead. if I didn't have that mindset in mind then I would have gone your route instead of simply making the collection into an array, though I would have done 
```js
if (!((roles instanceof Collection) || Array.isArray(roles))) return //error
//then
roles = roles.map(role=> this.guild.roles.resolve(role);
```
since collection and array both have a map function you can ditch the creation of an array from collection values and move straight to mapping, if you think it's better to do it your way then i would happy to switch to straight arrays